### PR TITLE
Feature/dynamic xpath changes

### PR DESCRIPTION
**Pull request by njerfia**
1. Removed src->types->d.ts file in Framework level. 
2. Changed script for Dynamic xpath in POM based on review comments

### CODE REVIEW

File: src/main/helper/pageHelper.ts [hunk 1]
```diff
@@ -227,23 +230,38 @@ module.exports = {
...
                const yLocation = await (await $(this.selector)).getLocation('y')
-               console.log(yLocation);
 
-               browser.moveToElement('$(this.selector)', xLocation, yLocation)
+               console.log('ylocation********** :',yLocation);
+
+               
+
+               this.moveTo(xLocation, yLocation)
+
```
Comment (balualtimetrik) [inline src/main/helper/pageHelper.ts:255]: please remove unnecessary spaces.

File: src/main/helper/pageHelper.ts [hunk 2]
```diff
@@ -227,23 +230,38 @@ module.exports = {
...
           } catch (error) {
+
                addLogger(`ERROR : ${error}`)
+
           }
+
```
Comment (balualtimetrik) [inline src/main/helper/pageHelper.ts:261]: please remove unnecessary spaces.

File: src/main/pageobjects/amz.productDetails.page.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+        await this.pdtImage.waitTillInterval(3000)
+        await (await this.pdtImage).isDisplayed()
+        await this.pdtImage.elementDoubleClick()
+        await this.fullScreenIMG.waitTillInterval(3000)
+        await (await this.fullScreenIMG).waitForDisplayed()
+        await this.fullScreenIMG.waitTillInterval(3000)
+        await this.ZoomIMGTitle.highlightElement()
+        await this.fullScreenIMG.waitTillInterval(3000)
+
```
Comment (balualtimetrik) [inline src/main/pageobjects/amz.productDetails.page.ts:79]: If these are test case steps, add them in the spec files. Page classes should have only locator and their actions. If you want to use re-usable methods, create a separate class under helper folder and import them in specs

File: test/specs/amz.productDetails.specs.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    });
+
+    it('Validate EMI Options', async () => {
+
+        await ProductDetailsPage.viewEMIOptions();
+
+    });
+
+    it('Product zoom', async () => {
+
```
Comment (balualtimetrik) [inline test/specs/amz.productDetails.specs.ts:48]: give a descriptive test case names.

Comment (njerfia) [reply test/specs/amz.productDetails.specs.ts:48]: Renamed Testcases descriptively, will reflect after commit

File: src/main/helper/pageHelper.ts (hunk 1 above)
Comment (njerfia) [reply src/main/helper/pageHelper.ts:255]: Formatted the file

File: src/main/helper/pageHelper.ts (hunk 2 above)
Comment (njerfia) [reply src/main/helper/pageHelper.ts:261]: Formatted the file

**Pull request closed without merging**